omod/src/main/java/org/openmrs/module/idgen/rest/resource/IdentifierSourceResource.java

```diff
@@ -0,0 +1,300 @@
+            description.addProperty("uuid");
+            description.addProperty("name");
+            description.addProperty("description");
+            description.addProperty("identifierType");
+
+            // links
+            description.addSelfLink();
+            if (rep instanceof DefaultRepresentation) {
+                description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+            }
+        }
+        return description;
+    }
+	
+    @Override
+    public DelegatingResourceDescription getCreatableProperties() {
+        DelegatingResourceDescription description = new DelegatingResourceDescription();
+        description.addProperty("name");
+        description.addProperty("description");
+        description.addProperty("identifierType");
+        return description;
+    }
+
+    @Override
+    public DelegatingResourceDescription getUpdatableProperties() throws ResourceDoesNotSupportOperationException {
+        DelegatingResourceDescription description = new DelegatingResourceDescription();
+        description.addProperty("name");
+        description.addProperty("description");
+        return description;
+    }
+
+    @PropertyGetter("identifierType")
+    public String getidentifierType(IdentifierSource identifierSource) {
+        return identifierSource.getIdentifierType().getUuid();
+    }
+	
+    @PropertyGetter("display")
+    public String getDisplayString(IdentifierSource identifierSource) {
+        return identifierSource.getIdentifierType() + " - " + identifierSource.getName();
+    }
+
+    @Override
+    public IdentifierSource newDelegate() {
+        return new SequentialIdentifierGenerator();
+    }
+	
+    @Override
+    public Object create(SimpleObject postBody, RequestContext context) throws ResponseException {
+		
+        Object savedIdentifierSource = null;
+        ArrayList<String> errors = new ArrayList<String>();
+        PatientIdentifierType patientIdentifierType = null;
+        Object identifierSourceType = postBody.get("sourceType");
+        Object identifierTypeUuid = postBody.get("identifierType");
+        Object name = postBody.get("name");
+        Object description = postBody.get("description");
+
+        if (identifierSourceType == null || StringUtils.isBlank(identifierSourceType.toString())) {
+            errors.add("source type"); 
+        }
+        if (name == null || StringUtils.isBlank(name.toString())) {
+            errors.add("name"); 
+        }
+        if (identifierTypeUuid == null || StringUtils.isBlank(identifierTypeUuid.toString())) {
+            errors.add("patient identifier type"); 
+        }else{
+            patientIdentifierType = Context.getPatientService().getPatientIdentifierTypeByUuid(identifierTypeUuid.toString());
+            if(patientIdentifierType == null){
+                errors.add("patient identifier type");
+            }
+        }
+        if(errors.toString().replaceAll("[\\[\\]]", "").length() > 0) {
```

Comment:
Am struggling to understand why you need to first call toString() is all you want to know is how many items are in this list

Reply:
replaced with: `if(errors.size() > 0) {}`